TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODBlackjackPlugin.cs

```diff
@@ -0,0 +1,2244 @@
+                            DealerIntermissionComplete = false;
+                            var firstEvent = DealerEventsQueue[0];
+                            DealerEventsQueue.RemoveAt(0);
+                            Controller.SendOBJEvent(firstEvent);
+                        }
+                        // if not, see if there are still players. goto betting if so, waiting if not
+                        else
+                        {
+                            if (!Lobby.IsEmpty())
+                                EnqueueGotoState(VMEODBlackjackStates.Betting_Round);
+                            else
+                                EnqueueGotoState(VMEODBlackjackStates.Waiting_For_Player);
+                        }
+                        break;
+                    }
+                case VMEODBlackjackStates.Waiting_For_Player:
+                    {
+                        ActivePlayerIndex = -1;
+                        CardDeck.Shuffle(2);
+                        GameState = newState;
+                        Controller.SendOBJEvent(new VMEODEvent((short)VMEODBlackjackEvents.Dealer_Collect_Cards));
+                        break;
+                    }
+                case VMEODBlackjackStates.Pending_Initial_Transactions:
+                    {
+                        Lobby.Broadcast("blackjack_timer", BitConverter.GetBytes(0));
+                        IdleTimer = 2;
+                        Tock = 0;
+                        GameState = newState;
+                        break;
+                    }
+                case VMEODBlackjackStates.Insurance_Prompts:
+                    {
+                        DecisionTimer = 10;
+                        Tock = 0;
+                        GameState = newState;
+                        for (int index = 0; index < 4; index++)
+                        {
+                            var player = Lobby.Players[index];
+                            if (player != null)
+                            {
+                                var slot = Lobby.GetSlotData(index);
+                                if (slot.BetAccepted)
+                                {
+                                    slot.PromptedForInsurance = true;
+                                    player.Send("blackjack_insurance_prompt", BitConverter.GetBytes(slot.BetAmount / 2));
+                                }
+                            }
+                        }
+                        break;
+                    }
+                case VMEODBlackjackStates.Finale:
+                    {
+                        int dealerTotal = 0; // send 0 if blackjack
+                        if (!DealerPlayer.CurrentHandType.Equals(VMEODBlackjackHandTypes.Blackjack))
+                            dealerTotal = DealerPlayer.CurrentHandTotal;
+                        Lobby.Broadcast("dealer_hand_total", BitConverter.GetBytes(dealerTotal));
+                        Tock = 0;
+                        IdleTi
```

Comment:
IndexOutOfRangeException? I don't like the presence of this function at all, and its use on the background thread. (as discussed)